src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -126,15 +127,17 @@ public function decode($data, $format, array $context = array())
                 $nbHeaders = $nbCols;
 
                 foreach ($cols as $col) {
-                    $headers[] = explode($keySeparator, $col);
+                    $headerChain = explode($keySeparator, $col);
+                    $headers[] = $headerChain;
+                    $headersDepth[] = count($headerChain);
                 }
 
                 continue;
             }
 
             $item = array();
             for ($i = 0; ($i < $nbCols) && ($i < $nbHeaders); ++$i) {
-                $depth = count($headers[$i]);
```

Comment:
I think we could improve performance even further by just prepending a backslash here: `\count($headers[$i])` (and revert the rest of the patch)
that would replace the function call by an opcode.

Reply:
tried this but didn't get any results :/
https://blackfire.io/profiles/da45d95b-9dbb-4f47-ad37-2273e0fd889f/graph

Reply:
OK, sorry, that's since for 7.2